addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as ts from 'typescript';
+import * as glob from 'glob';
+import * as path from 'path';
+
+/**
+ * Declare a MapType where key is a string and value is an array of type string
+ * Use for storing a file's path as key and its dependent module's relative path
+ */
+declare interface ModuleImport {
+  specifier: string;
+  pos: number;
+  end: number;
+};
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): ts.SourceFile {
+  return ts.createSourceFile(fileName,
+                             fs.readFileSync(fileName, 'utf-8'),
```

Comment:
@hansl, how do you feel about `readFileSync` in the CLI?

My concern would be that doing this - while convenient - precludes optimizing operations like this through concurrency in the future.